Actions/CheckForNewWarnings/CheckForNewWarnings.ps1

```diff
@@ -0,0 +1,49 @@
+Param(
+    [string] $token,
+    [string] $project,
+    [object] $settings,
+    [string] $targetBranch,
+    [string] $prBuildOutputFile
+)
+
+if (-not $ENV:GITHUB_BASE_REF)
+{
+    Write-Host "Checking for warnings only runs on pull requests."
+    return
+}
+
+if (-not $settings.failOnNewWarnings)
+{
+    Write-Host "Failing on new warnings is disabled in the settings."
+    return
+}
+
+try
+{
+    Write-Host "::group::Analyzing build for new warnings..."
+
+    Import-Module (Join-Path $PSScriptRoot "..\Github-Helper.psm1" -Resolve) -DisableNameChecking
+    Import-Module (Join-Path $PSScriptRoot ".\CheckForWarningsUtils.psm1" -Resolve) -DisableNameChecking
+
+    $baselineWorkflowRunId,$baselineWorkflowSHA = FindLatestSuccessfulCICDRun -repository $env:GITHUB_REPOSITORY -branch $targetBranch -token $token -retention $settings.incrementalBuilds.retentionDays
+    if ($project) { $projectName = $project } else { $projectName = $env:GITHUB_REPOSITORY -replace '.+/' }
```

Comment:
Is there a case when `$project` isn't set?

`RunPipeline` always runs in the context of a project. Default is `'.'`, which means the repo root is the AL-Go project, being built.